src/Commands/Category/CategoryProductsCount.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * Copyright (c) Since 2020 Friends of Presta
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * that is bundled with this package in the file docs/licenses/LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/afl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to infos@friendsofpresta.org so we can send you a copy immediately.
+ *
+ * @author    Friends of Presta <infos@friendsofpresta.org>
+ * @copyright since 2020 Friends of Presta
+ * @license   https://opensource.org/licenses/AFL-3.0  Academic Free License ("AFL") v. 3.0
+ *
+ */
+
+declare(strict_types=1);
+
+namespace FOP\Console\Commands\Category;
+
+use Category;
+use FOP\Console\Command;
+use PBergman\Console\Helper\TreeHelper;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class CategoryProductsCount extends Command
+{
+    /**
+     * @var LegacyContext
+     */
+    private $legacyContext;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure(): void
+    {
+        $this->setName('fop:category:products-count')
+            ->setDescription('Get the number of products for category and its children (root category by default)')
+            ->addUsage('2')
+            ->addUsage('--output=filename.csv')
+            ->addArgument('id-category', InputArgument::OPTIONAL, 'Category id')
+            ->addOption('output', 'o', InputOption::VALUE_REQUIRED, 'Write the output to a csv file');
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        parent::initialize($input, $output);
+
+        $this->legacyContext = $this->getContainer()->get('prestashop.adapter.legacy.context');
+    }
```

Comment:
you can also initialize the category object here, to avoid loading new instances each time you call the `getCategoryLabel` method for example.
something like : 
```php
$this->currentCategoryObject = new Categorie($idCat, $idLang, $idShop);
```